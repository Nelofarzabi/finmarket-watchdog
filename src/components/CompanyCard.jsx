import React from 'react';
import PropTypes from 'prop-types';
import { HiOutlineArrowRightCircle } from 'react-icons/hi2';

const CompanyCard = ({ name, price, symbol }) => (
  <li className="card text-white flex flex-col items-end justify-between h-[220px] md:h-[300px] bg-center bg-no-repeat bg-cover p-3 ">
    <div>
      <HiOutlineArrowRightCircle style={{ fontSize: '1.6rem' }} />
    </div>
    <h2 className=" flex flex-col gap-2">
      <p className="text-base md:text-2xl">
        {name.length > 14 ? `${name.slice(0, 10)}...` : name}
      </p>

      <span className="bg-slate-600 text-sm self-end rounded-md py-1 px-3 tracking-widest">{symbol}</span>
      <span className="self-end text-base md:text-2xl">
        $
        {price}
      </span>
    </h2>
  </li>
);

CompanyCard.propTypes = {
  name: PropTypes.string.isRequired,
  price: PropTypes.number.isRequired,
  symbol: PropTypes.string.isRequired,
};

export default CompanyCard;
