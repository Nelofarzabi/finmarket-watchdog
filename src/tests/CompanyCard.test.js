import React from 'react';
import { render } from '@testing-library/react';
import CompanyCard from '../components/CompanyCard';

describe('CompanyCard', () => {
  test('renders the CompanyCard component with provided props', () => {
    const props = {
      name: 'Apple',
      price: 100.5,
      symbol: 'APPL',
    };

    const { getByText } = render(
      <CompanyCard
        name={props.name}
        price={props.price}
        symbol={props.symbol}
      />,
    );

    // Check if the component renders the name, price, and symbol
    expect(getByText('Apple')).toBeInTheDocument();
    expect(getByText('APPL')).toBeInTheDocument();
    expect(getByText('$100.5')).toBeInTheDocument();
  });

  test('truncates long names with ellipsis', () => {
    const props = {
      name: 'Palisade Bio, Inc.',
      price: 200.75,
      symbol: 'PALI',
    };

    const { getByText } = render(
      <CompanyCard
        name={props.name}
        price={props.price}
        symbol={props.symbol}
      />,
    );

    // Check if the component truncates the long name with ellipsis
    expect(getByText('Palisade B...')).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const props = {
      name: 'Tesla, Inc',
      price: 150.25,
      symbol: 'TSLA',
    };

    const { container } = render(
      <CompanyCard
        name={props.name}
        price={props.price}
        symbol={props.symbol}
      />,
    );

    // Check if the rendered component matches the snapshot
    expect(container).toMatchSnapshot();
  });
});
