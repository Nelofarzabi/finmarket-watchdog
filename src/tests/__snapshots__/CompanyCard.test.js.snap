// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompanyCard matches snapshot 1`] = `
<div>
  <li
    class="card text-white flex flex-col items-end justify-between h-[220px] md:h-[300px] bg-center bg-no-repeat bg-cover p-3 "
  >
    <div>
      <svg
        aria-hidden="true"
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-width="1.5"
        style="font-size: 1.6rem;"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12.75 15l3-3m0 0l-3-3m3 3h-7.5M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
    </div>
    <h2
      class=" flex flex-col gap-2"
    >
      <p
        class="text-base md:text-2xl"
      >
        Tesla, Inc
      </p>
      <span
        class="bg-slate-600 text-sm self-end rounded-md py-1 px-3 tracking-widest"
      >
        TSLA
      </span>
      <span
        class="self-end text-base md:text-2xl"
      >
        $
        150.25
      </span>
    </h2>
  </li>
</div>
`;
